polyfill/lib/ecmascript.mjs

```diff
@@ -1352,10 +1352,9 @@ export function ToTemporalTime(item, options = undefined) {
   return CreateTemporalTime(time);
 }
 
-export function ToTemporalTimeOrMidnight(item) {
-  const TemporalPlainTime = GetIntrinsic('%Temporal.PlainTime%');
-  if (item === undefined) return new TemporalPlainTime();
-  return ToTemporalTime(item);
+export function ToTimeRecordOrMidnight(item) {
+  if (item === undefined) return MidnightTimeRecord();
+  return GetSlot(ToTemporalTime(item), TIME);
```

Comment:
Future work: it seems like `ToTemporalTime` could also return a record

Reply:
I did look at doing that, but I think I still prefer what it currently does; it's analogous with ToTemporalDate, ToTemporalZonedDateTime, etc. In this one case it'd be better to return a record, but I'm not sure if it'd be better elsewhere.